foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/util/YamlUtils.java

```diff
@@ -193,6 +186,36 @@ public Collection<Object> values() {
         public Set<Map.Entry<String, Object>> entrySet() {
             return contents.entrySet();
         }
+
+        private <T> boolean hasValue(Class<T> type, String... path) {
+            YamlObject parent = getParentIfExists(path);
+            if (parent == null) {
+                return false;
+            }
+            String key = path[path.length - 1];
+            return parent.containsKey(key) && type.isInstance(parent.get(key));
+        }
+
+        private <T> T getValue(String... path) {
```

Comment:
I: An argument could be made that we only need these two type-parameterized methods and the `hasObject/getObject` methods, but it can be difficult to understand how to work with an API that only has type-parameterized methods.